File: lab-1/c#/Triangle.cs
using System;

namespace TriangleApp
{
    public class Triangle
    {
        private double x1, y1;
        private double x2, y2;
        private double x3, y3;

        // Конструктор за замовчуванням
        public Triangle()
        {
            x1 = y1 = x2 = y2 = x3 = y3 = 0.0;
        }

        // Конструктор з параметрами
        public Triangle(double x1, double y1, double x2, double y2, double x3, double y3)
        {
            this.x1 = x1; 
            this.y1 = y1;
            this.x2 = x2; 
            this.y2 = y2;
            this.x3 = x3; 
            this.y3 = y3;
        }

        // Метод для встановлення координат
        public void SetCoordinates(double x1, double y1, double x2, double y2, double x3, double y3)
        {
            this.x1 = x1;
            this.y1 = y1;
            this.x2 = x2;
            this.y2 = y2;
            this.x3 = x3;
            this.y3 = y3;
        }

        // Гетери (властивості) для читання координат
        public double X1 => x1;
        public double Y1 => y1;
        public double X2 => x2;
        public double Y2 => y2;
        public double X3 => x3;
        public double Y3 => y3;

        // Метод обчислення периметра
        public double GetPerimeter()
        {
            double sideA = Math.Sqrt((x2 - x1) * (x2 - x1) + (y2 - y1) * (y2 - y1));
            double sideB = Math.Sqrt((x3 - x2) * (x3 - x2) + (y3 - y2) * (y3 - y2));
            double sideC = Math.Sqrt((x1 - x3) * (x1 - x3) + (y1 - y3) * (y1 - y3));

            return sideA + sideB + sideC;
        }

        // Метод обчислення площі (за формулою Герона)
        public double GetArea()
        {
            double sideA = Math.Sqrt((x2 - x1) * (x2 - x1) + (y2 - y1) * (y2 - y1));
            double sideB = Math.Sqrt((x3 - x2) * (x3 - x2) + (y3 - y2) * (y3 - y2));
            double sideC = Math.Sqrt((x1 - x3) * (x1 - x3) + (y1 - y3) * (y1 - y3));

            double p = (sideA + sideB + sideC) / 2.0; // півпериметр
            return Math.Sqrt(p * (p - sideA) * (p - sideB) * (p - sideC));
        }
    }
}

File: lab-1/c++/Triangle.cpp
#include "Triangle.h"
#include <cmath>  // для sqrt

// Конструктор за замовчуванням (задає координати 0,0,0,0,0,0)
Triangle::Triangle()
    : x1(0), y1(0), x2(0), y2(0), x3(0), y3(0)
{
}

// Конструктор з параметрами
Triangle::Triangle(double x1, double y1, double x2, double y2, double x3, double y3)
    : x1(x1), y1(y1), x2(x2), y2(y2), x3(x3), y3(y3)
{
}

// Сетери
void Triangle::setCoordinates(double X1, double Y1, double X2, double Y2, double X3, double Y3)
{
    x1 = X1; 
    y1 = Y1;
    x2 = X2; 
    y2 = Y2;
    x3 = X3; 
    y3 = Y3;
}

// Гетери
double Triangle::getX1() const { return x1; }
double Triangle::getY1() const { return y1; }
double Triangle::getX2() const { return x2; }
double Triangle::getY2() const { return y2; }
double Triangle::getX3() const { return x3; }
double Triangle::getY3() const { return y3; }

// Обчислення периметра
double Triangle::getPerimeter() const
{
    // Довжини сторін
    double sideA = std::sqrt((x2 - x1) * (x2 - x1) + (y2 - y1) * (y2 - y1));
    double sideB = std::sqrt((x3 - x2) * (x3 - x2) + (y3 - y2) * (y3 - y2));
    double sideC = std::sqrt((x1 - x3) * (x1 - x3) + (y1 - y3) * (y1 - y3));

    return sideA + sideB + sideC;
}

// Обчислення площі (за формулою Герона)
double Triangle::getArea() const
{
    double sideA = std::sqrt((x2 - x1) * (x2 - x1) + (y2 - y1) * (y2 - y1));
    double sideB = std::sqrt((x3 - x2) * (x3 - x2) + (y3 - y2) * (y3 - y2));
    double sideC = std::sqrt((x1 - x3) * (x1 - x3) + (y1 - y3) * (y1 - y3));

    double p = (sideA + sideB + sideC) / 2.0; // півпериметр
    return std::sqrt(p * (p - sideA) * (p - sideB) * (p - sideC));
}
